smack-experimental/src/main/java/org/jivesoftware/smackx/mam/MamManager.java

```diff
@@ -0,0 +1,533 @@
+            XMPPErrorException, NotConnectedException, InterruptedException, NotLoggedInException {
+        return queryArchive(null, null, null, null, additionalFields);
+    }
+
+    /**
+     * Query archive filtering by start date. The value of 'end' will be equal
+     * to the date/time of the most recent message stored in the archive.
+     * 
+     * @param start
+     * @return the MAM query result
+     * @throws NoResponseException
+     * @throws XMPPErrorException
+     * @throws NotConnectedException
+     * @throws InterruptedException
+     * @throws NotLoggedInException
+     */
+    public MamQueryResult queryArchiveWithStartDate(Date start) throws NoResponseException, XMPPErrorException,
+            NotConnectedException, InterruptedException, NotLoggedInException {
+        return queryArchive(null, start, null, null, null);
+    }
+
+    /**
+     * Query archive filtering by end date. The value of 'start' will be equal
+     * to the date/time of the earliest message stored in the archive.
+     * 
+     * @param end
+     * @return the MAM query result
+     * @throws NoResponseException
+     * @throws XMPPErrorException
+     * @throws NotConnectedException
+     * @throws InterruptedException
+     * @throws NotLoggedInException
+     */
+    public MamQueryResult queryArchiveWithEndDate(Date end) throws NoResponseException, XMPPErrorException,
+            NotConnectedException, InterruptedException, NotLoggedInException {
+        return queryArchive(null, null, end, null, null);
+    }
+
+    /**
+     * Query archive applying filters: max count, start date, end date, from/to
+     * JID and with additional fields.
+     * 
+     * @param max
+     * @param start
+     * @param end
+     * @param withJid
+     * @param additionalFields
+     * @return the MAM query result
+     * @throws NoResponseException
+     * @throws XMPPErrorException
+     * @throws NotConnectedException
+     * @throws InterruptedException
+     * @throws NotLoggedInException
+     */
+    public MamQueryResult queryArchive(Integer max, Date start, Date end, String withJid,
+            List<AdditionalField> additionalFields) throws NoResponseException, XMPPErrorException,
+            NotConnectedException, InterruptedException, NotLoggedInException {
+        DataForm dataForm = null;
+        String queryId = UUID.randomUUID().toString();
+
+        if (start != null || end != null || withJid != null || additionalFields != null) {
+            dataForm = getNewMamForm();
+            addStart(start, dataForm);
+            addEnd(end, dataForm);
+            addWithJid(withJid, dataForm);
+            addAdditionalFields(additionalFields, dataForm);
+        }
+
+        MamQueryIQ mamQueryIQ = prepareMamQueryIQSet(dataForm, queryId);
+
+        addResultsLimit(max, mamQueryIQ);
+
+        return queryArchive(mamQueryIQ, 0);
+    }
+
+    private void addAdditionalFields(List<AdditionalField> additionalFields, DataForm dataForm) {
```

Comment:
You think in the wrong direction. ;) It is not necessary to make them non-static i.e. virtual methods.

Instead of
```java
static void eventuallyDoSomething(Foo foo, Bar bar) {
  if (foo != null) {
      foo.someting(bar);
  }
}
```
you should write
```java
static void eventuallyDoSomething(Foo foo, Bar bar) {
  if (foo == null) return;
  foo.something(bar);
}
```